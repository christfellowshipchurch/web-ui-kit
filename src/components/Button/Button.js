import React from 'react'
import classnames from 'classnames'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faAngleRight } from '@fortawesome/pro-light-svg-icons'

const Button = ({
  size,
  type,
  call,
  action
}) => {

  if (type === "dark") {
    type = "outline-dark";
  } else if (type === "light") {
    type = "outline-light";
  } else if (type !== 'link' && type !== 'primary') {
    type = 'primary'
  }

  const classes = classnames(
    'btn',
    size !== '' ? `btn-${size}` : '',
    `btn-${type}`,
    type !== 'link' ? 'text-uppercase' : '',
    'font-weight-bold',
  )

  return (
    <button onClick={action} className={classes}>
      {call} {type === 'link' && <FontAwesomeIcon icon={faAngleRight} className="ml-1 mb-0" />}
    </button>
  )
}

Button.defaultProps = {
  size: '',
  type: 'primary',
  call: 'default',
  action: () => { }
}


export default Button