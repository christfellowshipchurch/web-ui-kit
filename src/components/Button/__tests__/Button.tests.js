import React from 'react'
import Button from '..'
import renderer from 'react-test-renderer'

describe('Button', () => {
    // Deprecated Props
    it('renders a Default Button with deprecated properties', () => {
        const tree = renderer.create(
            <Button
                title="Button Title"
                onClick={() => true}
            />
        )

        expect(tree).toMatchSnapshot()
    })

    // Button Types
    it('renders a Default Button without crashing', () => {
        const tree = renderer.create(
            <Button
                title="Button Title"
                onClick={() => true}
            />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Link Button without crashing', () => {
        const tree = renderer.create(
            <Button
                type="link"
                title="Button Title"
                onClick={() => true}
            />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Light Button without crashing', () => {
        const tree = renderer.create(
            <Button
                type="light"
                title="Button Title"
                onClick={() => true}
            />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Dark Button without crashing', () => {
        const tree = renderer.create(
            <Button
                type="dark"
                title="Button Title"
                onClick={() => true}
            />
        )

        expect(tree).toMatchSnapshot()
    })

    // Button Size
    it('renders a Button with a size override', () => {
        const tree = renderer.create(
            <Button
                size="lg"
                title="Button Title"
                onClick={() => true}
            />
        )

        expect(tree).toMatchSnapshot()
    })
})