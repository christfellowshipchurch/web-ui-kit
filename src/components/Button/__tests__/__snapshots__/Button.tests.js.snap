// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a Button with a size override 1`] = `
<button
  className="btn btn-lg btn-primary text-uppercase"
  onClick={[Function]}
>
  Button Title
</button>
`;

exports[`Button renders a Dark Button without crashing 1`] = `
<button
  className="btn btn-outline-dark text-uppercase"
  onClick={[Function]}
>
  Button Title
</button>
`;

exports[`Button renders a Default Button with deprecated properties 1`] = `
<button
  className="btn btn-primary text-uppercase"
  onClick={[Function]}
>
  Button Title
</button>
`;

exports[`Button renders a Default Button without crashing 1`] = `
<button
  className="btn btn-primary text-uppercase"
  onClick={[Function]}
>
  Button Title
</button>
`;

exports[`Button renders a Light Button without crashing 1`] = `
<button
  className="btn btn-outline-light text-uppercase"
  onClick={[Function]}
>
  Button Title
</button>
`;

exports[`Button renders a Link Button without crashing 1`] = `
<button
  className="btn btn-link"
  onClick={[Function]}
>
  Button Title
</button>
`;
